src/lib/replication/repl_dev/common.cpp

```diff
@@ -199,7 +212,7 @@ std::string repl_req_ctx::to_compact_string() const {
 }
 
 bool repl_req_ctx::is_expired() const {
-    return get_elapsed_time_sec(m_start_time) > HS_DYNAMIC_CONFIG(consensus.repl_req_timeout_sec);
+    return time_from(repl_req_state_t::LOG_RECEIVED) > HS_DYNAMIC_CONFIG(consensus.repl_req_timeout_sec);
```

Comment:
if log received, before follower received the data (rreq will also be created), originator is gone, we should also clean up the rreqs, right?

Reply:
yes right now (the PR) is trying to be compatible with original implementation for `is_expired`,  mainly to bring in all the timestamp we have as a foundation  of next step 

I think we discussed the GC stuff a bit today and that need a seperate design, we can visit that later.